File: templates/lang/android_app_dagger/src_files/main/MainActivityModule.java
package {{ .Group }}.main;

import android.app.Activity;

import {{ .Group }}.base.BaseActivityModule;

import com.episode6.hackit.android.inject.context.scope.ActivityScope;

import dagger.Binds;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module(includes = {BaseActivityModule.class})
public abstract class MainActivityModule {

  @Binds
  @ActivityScope
  abstract Activity bindMainActivity(MainActivity activity);

  @ContributesAndroidInjector
  abstract MainFragment mainFragment();
}

File: templates/lang/kotlin_android_app_dagger/src_files/main/MainActivityModule.kt
package {{ .Group }}.main

import android.app.Activity
import com.episode6.hackit.android.inject.context.scope.ActivityScope
import com.episode6.hackit.android.inject.context.scope.FragmentScope
import {{ .Group }}.base.BaseActivityModule
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module(includes = arrayOf(BaseActivityModule::class))
abstract class MainActivityModule {

  @Binds
  @ActivityScope
  internal abstract fun bindMainActivity(activity: MainActivity): Activity

  @FragmentScope
  @ContributesAndroidInjector
  internal abstract fun mainFragment(): MainFragment
}
